### PR TITLE
[IMP] stock_account: Allow tracking COGS Journal Items origin

### PR DESCRIPTION
**Pull request by hbto**
Main
-

[IMP] stock_account: Allow to keep track of the COGS Journal Items origin

Explain
-

So far It is quite cumbersome to try to figure out which lines are coming from where for the COGS journal items. It becomes more difficult to find them out when there are several lines from the same product with the same quantities and similar characteristics.

By having what invoice line originated which COGS Journal Items you can auditing and analysis in a more thorough way.

- You can figure out which Sale Line o Invoice Line was more profitable. Based on the actual revenue and the actual COGS.
- What COGS line was mistakenly created. For instance, when the invoice is created prior delivery, you can figure out how to surgically fix the of COGS journal items that that were wrong.
- To try to reconcile properly the lines in a Sale Order Line when there is drift and reconciliation is not properly made because there were differences and they cannot be easily spotted because you get lost in middle of several COGS lines but you do not know where they are really coming from.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hbto** [comment]
Hello @william-andre this is the PR I have talked about in https://github.com/odoo/odoo/pull/124574

Best Regards.

**Comment by hbto** [comment]
cc/ @nhomar @jtijerino @luisg123v

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: addons/stock_account/models/account_move.py [hunk 1]
```diff
@@ -228,6 +230,7 @@ class AccountMoveLine(models.Model):
     _inherit = 'account.move.line'
 
     stock_valuation_layer_ids = fields.One2many('stock.valuation.layer', 'account_move_line_id', string='Stock Valuation Layer')
+    origin_of_cogs_aml_id = fields.Many2one("account.move.line", copy=False, help="Technical field used to keep track in the originating line of the anglo-saxon lines.")
```
[the remark this answers is not in the document]
Comment (hbto) [reply addons/stock_account/models/account_move.py]: @luisg123v Taken care of this.

Regards.

**Comment by hbto** [comment]
@william-andre Do you have any feedback regarding this request?

Best Regards.

File: addons/stock_account/models/account_move.py [hunk 2]
```diff
@@ -232,6 +234,9 @@ class AccountMoveLine(models.Model):
     _inherit = 'account.move.line'
 
     stock_valuation_layer_ids = fields.One2many('stock.valuation.layer', 'account_move_line_id', string='Stock Valuation Layer')
+    origin_of_cogs_aml_id = fields.Many2one(
+        "account.move.line", copy=False,
+        help="Technical field used to keep track in the originating line of the anglo-saxon lines.")
```
Comment (william-andre) [inline addons/stock_account/models/account_move.py:239]:
```suggestion
    cogs_origin_id = fields.Many2one(  # technical field used to keep track in the originating line of the anglo-saxon lines
        "account.move.line", copy=False,
    )
```

* `help` is intended for end users, not for devs. Putting tech info in there make the UI heavier
* close the parenthesis on the same indentation level than its opening

**Review by william-andre — COMMENTED** [review]
Sorry for the delay, I was on leave for a long time, and had a big deadline before that.

This seems okay :+1:

File: addons/stock_account/models/account_move.py (hunk 2 above)
Comment (hbto) [reply addons/stock_account/models/account_move.py]: @william-andre Thanks for the reply.

I have applied your suggestions.

Could please review again.

Best Regards.

**Review by william-andre — APPROVED** [review]
@robodoo r+

Thank you for the contribution!

**Pull request closed without merging**